src/modules/validation/abnf.ts

```diff
@@ -38,10 +38,12 @@ export class ValidatedUrl {
   }
 
   public validate(graphUrl: string): ValidationResult {
+    let decodedGraphUrl = graphUrl;
```

Comment:
Can we please have some tests to validate that we're fixing the issues and not introducing new ones? Or at least one that validates that this fixes an issue?

Why is decodeURI being called? are there values in the encoded version of the string which might cause issues when fed into the parser here?

Reply:
Yes. The parser requires that the URL be passed in without encoding. Passing a lone % to the parser would cause it to fail the decode step. 

Let me add more tests

Reply:
@gavinbarron take a look